platform/windows/display_server_windows.cpp

```diff
@@ -3482,9 +3482,11 @@ DisplayServerWindows::~DisplayServerWindows() {
 		SystemParametersInfoA(SPI_SETMOUSETRAILS, restore_mouse_trails, 0, 0);
 	}
 #ifdef GLES3_ENABLED
-	if (gl_manager) {
-		memdelete(gl_manager);
-		gl_manager = nullptr;
+	if (rendering_driver == "opengl3") {
+		if (gl_manager) {
+			memdelete(gl_manager);
+			gl_manager = nullptr;
+		}
 	}
```

Comment:
The logic around this may have changed quite a bit since I originally wrote it (and it probably wasn't all that great to begin with :grinning: ). I think the idea was that if a `gl_manager` _has been_ created by this point, you want to delete it, rather than leak it, which is the alternative.

This new logic seems to be assuming that `if rendering_driver != "opengl3"` it should have never been created, so maybe an assert type condition could be better (as the result should be the same in either case if you assumption was correct)?

Reply:
@lawnjelly it is definately worth discussing which direction we take here. This is why I set the default to a nullptr. I'm always in favor of checking that. If it wasn't created it will be null, if it was it will have a valid pointer.

The rendering_driver check I mostly added because it seems to be checking it that way in various places already.